### PR TITLE
Added Stack data structure

### PR DESCRIPTION
**Pull request by Kartikei-12**
Stack data structure created and some common method like push, pop added

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -4,3 +4,5 @@ __pycache__/
 *.pyo
 .vscode/
 .pytest_cache/
+.vscode
```
Comment (czgdp1807) [inline .gitignore]: `.vscode/` is already there. Any specific reasons for adding this once again?

File: .travis.yml [hunk 2]
```diff
@@ -2,6 +2,7 @@ language: python
 python:
         - "2.7"
         - "3.5"
+        - "3.6"
```
Comment (czgdp1807) [inline .travis.yml]: We have no plans of supporting `3.6` currently.

File: pydatastructs/linear_data_structures/stack.py [hunk 3]
```diff
@@ -0,0 +1,98 @@
+"""stack.py file containning defination of 'Stack' datastructure"""
```
Comment (czgdp1807) [inline pydatastructs/linear_data_structures/stack.py]: This can be removed.

File: pydatastructs/linear_data_structures/stack.py [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+
+from __future__ import print_function, division
+from copy import deepcopy as dc
+
+from ..utils.type_utils import _check_type
+
+__all__ = ["Stack"]
```
Comment (czgdp1807) [inline pydatastructs/linear_data_structures/stack.py]:
```python
__all__ =  [
    'Stack'
]
```

File: pydatastructs/utils/type_utils.py [hunk 5]
```diff
@@ -0,0 +1,4 @@
+"""type_utils.py file containning 'type' related utils"""
+
+_check_type = lambda a, t: isinstance(a, t)
+NoneType = type(None)
```
Comment (czgdp1807) [inline pydatastructs/utils/type_utils.py]: Add a new line.

File: pydatastructs/utils/type_utils.py [hunk 6]
```diff
@@ -0,0 +1,4 @@
+"""type_utils.py file containning 'type' related utils"""
```
Comment (czgdp1807) [inline pydatastructs/utils/type_utils.py]: This can be removed.

File: pydatastructs/linear_data_structures/stack.py [hunk 7]
```diff
@@ -0,0 +1,98 @@
...
+        element = dc(self.stack[-1])
+        del self.stack[-1]
+        return element
+
+    def __len__(self):
+        """Provides length of Stack"""
+        return len(self.stack)
+
+    def __str__(self):
+        """__str__ Method"""
```
Comment (czgdp1807) [inline pydatastructs/linear_data_structures/stack.py]:
```python
"""
Used for printing.
"""
```

File: pydatastructs/linear_data_structures/stack.py [hunk 8]
```diff
@@ -0,0 +1,98 @@
...
+        return "<Stack length:{}>".format(len(self.stack))
```
Comment (czgdp1807) [inline pydatastructs/linear_data_structures/stack.py]: Instead of length, it is preferred to return `list` casted to `str`.

**Comment by czgdp1807** [comment]
**General comments**

- Please follow the PR template, with references to the relevant literature(preferably Wikipedia).

- Follow the [numpy doc format](https://numpydoc.readthedocs.io/en/latest/format.html).

File: .travis.yml (hunk 2 above)
Comment (Kartikei-12) [reply .travis.yml]: Got it, I was working with 3.6 so I added will resolve any issues.

File: pydatastructs/linear_data_structures/stack.py (hunk 3 above)
Comment (Kartikei-12) [reply pydatastructs/linear_data_structures/stack.py]: It is suggested as doc generator like sphinx add this to file description which help in understanding files structure when using documentation

File: pydatastructs/linear_data_structures/stack.py (hunk 4 above)
Comment (Kartikei-12) [reply pydatastructs/linear_data_structures/stack.py]: I am using [black]() for code formatting, handy tool, any specific reason to use above format.

File: .gitignore (hunk 1 above)
Comment (Kartikei-12) [reply .gitignore]: Just a small oversight, corrected.

File: pydatastructs/linear_data_structures/stack.py (hunk 4 above)
Comment (czgdp1807) [reply pydatastructs/linear_data_structures/stack.py]: We are following the coding style of `SymPy`, so please follow that.

**Comment by czgdp1807** [comment]
Why closed?

**Comment by Kartikei-12** [comment]
Making new PR with appropriate changes.

**Pull request closed without merging**